tests/objectregistry/server.cpp

```diff
@@ -0,0 +1,85 @@
+/**********************************************************************
+**
+** Copyright (C) 2020 Luxoft Sweden AB
+**
+** This file is part of the FaceLift project
+**
+** Permission is hereby granted, free of charge, to any person
+** obtaining a copy of this software and associated documentation files
+** (the "Software"), to deal in the Software without restriction,
+** including without limitation the rights to use, copy, modify, merge,
+** publish, distribute, sublicense, and/or sell copies of the Software,
+** and to permit persons to whom the Software is furnished to do so,
+** subject to the following conditions:
+**
+** The above copyright notice and this permission notice shall be
+** included in all copies or substantial portions of the Software.
+**
+** THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+** EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+** MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+** NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS
+** BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+** ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+** CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+** SOFTWARE.
+**
+** SPDX-License-Identifier: MIT
+**
+**********************************************************************/
+#include "server.h"
+#include "common.h"
+#include <QCoreApplication>
+#include <QProcess>
+
+namespace tests {
+namespace ipc {
+
+OtherIPCTestInterfaceImpl::OtherIPCTestInterfaceImpl()
+{
+    m_adapter.setEnabled(true);
+    m_adapter.registerService(this);
+}
+
+void OtherIPCTestInterfaceImpl::requestExit()
+{
+    emit exitRequested();
+    qDebug() << "Test passed";
+    QCoreApplication::quit();
```

Comment:
Why not exiting gracefully with the following ? :
qApp->exit()

Reply:
I don't understand your question.
https://doc.qt.io/qt-5/qcoreapplication.html#quit
https://doc.qt.io/qt-5/qcoreapplication.html#exit
So qApp->exit() is exactly the same.

Reply:
you are right.